atom/browser/ui/file_dialog_mac.mm

```diff
@@ -93,6 +93,7 @@ void SetupDialog(NSSavePanel* dialog,
 
 void SetupDialogForProperties(NSOpenPanel* dialog, int properties) {
   [dialog setCanChooseFiles:(properties & FILE_DIALOG_OPEN_FILE)];
+  [dialog setTreatsFilePackagesAsDirectories:YES];
```

Comment:
Yes adding the option instead of making it mandatory would be better. Made the changes.